GameFiles/_integration.gsc

```diff
@@ -278,27 +297,71 @@ RegisterClientCommands()
 
 InitializeGameMethods() 
 {
-    level.overrideMethods = [];
-    level.overrideMethods["god"] = ::_god;
-    level.overrideMethods["noclip"] = ::UnsupportedFunc;
-    
-    if ( isDefined( ::God ) )
+    level.overrideMethods["UnsupportedFunc"]= ::UnsupportedFunc;
+    level notify("InitializeGameMethods");
+    level waittill("InitializeGameMethodsDone");
+}
+
+// Not every game can output to console or even game log.
+// Adds a very basic logging system that every
+// game specific script can extend.accumulate
+// Logging to dvars used as example.
+InitializeLogger()
+{
+    level.logger._logger = [];
+    RegisterLogger( ::Log2Dvar );
+    RegisterLogger( ::Log2IngamePrint );
+    level.logger.warning = ::LogWarning;
+    level.logger.error = ::LogError;
+}
+
+Log( LogLevel, message )
+{
+    for( i = 0; i < level.logger._logger.size; i++ )
     {
-        level.overrideMethods["god"] = ::God;
+        [[level.logger._logger[i]]]( LogLevel, message );
     }
-    
-    if ( isDefined( ::NoClip ) )
-    {
-        level.overrideMethods["noclip"] = ::NoClip;
+}
+
+LogWarning( message )
+{
+    Log( "warning", message );
+}
+
+LogError( message )
+{
+    Log( "error", message );
+}
+
+Log2Dvar( LogLevel, message )
+{
+    switch (LogLevel) {
```

Comment:
format: ( LogLevel )
inconsistent brace new line

Reply:
I would like you to use 'createprintchannel' builtin to create different channels for the various levels of logging.
Do not use IPrintLn. Use or printf

Reply:
Will that work on all cod titles? _integration.gsc should be as generic as possible so it can be used in as many cod titles as possible without changes. New loggers that use game specific functionality can always be registered in the game specific files.

Reply:
print will work on T5, T6 andIW5.
On T4/IW4 it will only work with developer_script set to 1. Alternative: printconsole.
createprintchannel and setprintchannel are as generic as it gets.
See utility.gsc